src/http2_frame_priority.erl

```diff
@@ -2,27 +2,54 @@
 
 -include("http2.hrl").
 
--export([
-         format/1,
-         read_binary/2,
-         read_priority/1,
-         to_binary/1
-        ]).
+-export(
+   [
+    format/1,
+    new/3,
+    stream_id/1,
+    read_binary/2,
+    read_priority/1,
+    to_binary/1
+   ]).
+
+-record(priority, {
+    exclusive = 0 :: 0 | 1,
+    stream_id = 0 :: stream_id(),
+    weight = 0 :: non_neg_integer()
+  }).
+-type payload() :: #priority{}.
+-export_type([payload/0]).
 
 -behaviour(http2_frame).
 
--spec format(priority()) -> iodata().
+-spec format(payload()) -> iodata().
 format(Payload) ->
     io_lib:format("[Priority: ~p]", [Payload]).
```

Comment:
someday we should make these look nice. this is just gonna be `{priority, 17, 3, 5}`, or whatever